content/automate/ManageIQ/System/Event/EmsEvent/RHEVM.class/__methods__/update_migration_status.rb

```diff
@@ -0,0 +1,54 @@
+module ManageIQ
+  module Automate
+    module System
+      module Event
+        module EmsEvent
+          module RHEVM
+            class UpdateMigrationStatus
+              def initialize(handle = $evm)
+                @handle = handle
+              end
+
+              def main
+                update_migration_status(vm)
+              end
+
+              private
+
+              def vm_id
+                event_stream = @handle.root['event_stream']
+                if event_stream.nil?
+                  @handle.log(:error, 'event_stream not found')
+                  raise 'event_stream not found'
+                end
+                event_stream.vm_or_template_id
+              end
+
+              def vm
+                @handle.vmdb('Vm', vm_id).tap do |vm|
+                  if vm.nil?
+                    @handle.log(:error, 'VM object not found')
+                    raise 'VM object not found'
+                  end
+                end
+              end
+
+              def migration_status
+                @handle.root['event_type'] == 'VM_MIGRATION_FAILED_FROM_TO' ? 'failure' : 'success'
+              end
+
+              def update_migration_status(vm)
+                @handle.log(:info, "Updating VM [#{vm.name}] migration status to [#{migration_status}]")
+                vm.set_migration_status(migration_status)
```

Comment:
`set_migration_status` method can be exposed in RHEV's VM class, but not the base class.

Reply:
If we want to not to have default implementation we need to check whether the method exists.